calculator_with_enhancements4.rb

```diff
@@ -0,0 +1,116 @@
+
+# create methods to define 4 operators
+
+def add(x,y)
+  return x+y
+end
+
+def subtract(x,y)
+  return x-y
+end
+
+def multiply(x,y)
+  return x*y
+end
+
+def divide(x,y)
+  return x/y
+end
+
+def exponent(x,y)
+  return x**y
+end
+
+def modulo(x,y)
+  return x%y
+end
+
+def valid_numeric_input(x)
+  input_to_integer = x.to_i
+  integer_back_to_string = input_to_integer.to_s
+  if x == "0" || x == integer_back_to_string
+    return true
+  elsif x == "" || x != integer_back_to_string
+    return false
+  else
+    return "error"
+  end
+end
+
+# asks user for the type of operation
+valid_operator = false
+
+while valid_operator == false
+  puts "What type of calculation do you want to do? Please enter one of the following options:
+    add or +
+    subtract or -
+    multiply or x or *
+    divide or /
+    exponent or **
+    modulo or %"
+
+    operation = gets.chomp.downcase
+
+  if operation == "add" || operation ==  "+" || operation == "subtract" || operation == "-" || operation == "multiply" || operation == "x" || operation == "*" || operation == "divide" || operation == "/" || operation == "modulo" || operation == "%"
```

Comment:
doesn't include exponent or **! Which was an optional requirement, not a big deal, but your intro says that it accepts those operators